clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/TransferMetadata.scala

```diff
@@ -71,4 +71,12 @@ trait TransferMetadata[M <: TransferMetadata[M]] { self: M =>
       case Some(existing) => Some(s"$existing\n$note")
       case None           => Some(note)
     }
+
+  /**
+    * Return a copy of this in which the `URI` vals are prefixed with
+    * `newPrefix` as necessary
+    *
+    * @return new metadata with some field values prefixed.
+    */
+  def prefixed(newPrefix: String): M = { val _ = newPrefix; self }
```

Comment:
Instead of a new method that all the subclasses need to implement, could we add a new `Option[String]` parameter to `moveInto` and `moveIntoDirectory` that injects the new prefix along with changing the leading path? That way we can roll the functionality out to all existing subclasses without needing more boilerplate.